### ISSUE TITLE
SC1036, SC1088: Eval of array with nested command substitution invalid, but works?

### ISSUE THREAD

**Issue by memotype** [koalaman/shellcheck#1615]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1088
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

ex='"line 1"
"line 2"
"line 3"'

eval lines=($(echo "$ex"))

for line in "${lines[@]}"; do
  echo "x${line}x"
done
```

#### Here's what shellcheck currently says:

```
Line 3:
ex='"line 1"
^-- SC2034: ex appears unused. Verify use (or export if used externally).
 
Line 7:
eval lines=($(echo "$ex"))
           ^-- SC1036: '(' is invalid here. Did you forget to escape it?
           ^-- SC1088: Parsing stopped here. Invalid use of parentheses?
           ^-- SC1098: Quote/escape special characters when using eval, e.g. eval "a=(b)".
```

#### Here's what I wanted or expected to see:

SC2034 is just a warning, but it is odd that shellcheck thinks it's unused... However, the really odd part is that SC1036 and SC 1088 are both hard errors (not warnings) and both seem to think that this is straight-up invalid syntax, but this code works just fine. I know SC1098 suggests quoting the full eval, which is probably best practice, but the code as presented works without any issues, regardless of the previous shellcheck errors.

```
$ ./foo.sh
xline 1x
xline 2x
xline 3x
```

The real-world use-case here, in case you're wondering, is that I'm using BitBucket API calls with `curl`, and passing them through `jq`, so I know each line of output is a quoted string, so each element of the resulting array is the name of a BitBucket repository, even if they have spaces in the name. The actual code I'm using is:

```sh
#!/bin/bash

user=redacted
pass=redacted
account=redacted

get() {
  local IFS=/
  base="$1"
  shift
  curl -su "$user:$pass" \
    "https://api.bitbucket.org/2.0/$base/$account/$*"
}

eval repos=($(get repositories | jq '.values[].name'))

# ...
```

The Output of the get() function will look like:

```
"Repository 1"
"Application Foo"
"etc-etc-42"
```

**Comment by koalaman** [comment]
[SC1098](https://github.com/koalaman/shellcheck/wiki/SC1098) mentions this, but I agree that ShellCheck should really parse it the way bash does even if it emits this warning.
